### PR TITLE
Search widgets

### PR DESCRIPTION
**Pull request by carolinux**
@m-kuhn please review

### CODE REVIEW

File: src/gui/editorwidgets/qgsvaluerelationsearchwidgetwrapper.cpp [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+{
+  if ( config( "AllowMulti" ).toBool() )
+  {
+    return new QgsFilterLineEdit( parent );
+  }
+  else if ( config( "UseCompleter" ).toBool() )
+  {
+    return new QgsFilterLineEdit( parent );
+  }
+  {
```
Comment (m-kuhn) [inline src/gui/editorwidgets/qgsvaluerelationsearchwidgetwrapper.cpp]: This looks strange, should that be an else?

Comment (carolinux) [inline src/gui/editorwidgets/qgsvaluerelationsearchwidgetwrapper.cpp]: It's copy pasted from qgsvaluerelationwidgetwrapper... I can change it in both places for style if you want, behavior would be the same.

Comment (m-kuhn) [inline src/gui/editorwidgets/qgsvaluerelationsearchwidgetwrapper.cpp]: Yes, that would make sense to me

**Comment by m-kuhn** [comment]
I have some rendering issues here:

screenshot from 2015-06-23 12 13 46

**Comment by carolinux** [comment]
steps taken to get this?

**Comment by m-kuhn** [comment]
I think you have to make sure that only one widget is visible at a given time.
Anyway, I think it will be better to postpone it to post 2.10, it feels to tight to merge it now. Let's just make sure that the current API is not exposed to python so this change can be merged without much worries.

**Comment by carolinux** [comment]
Now I can't reproduce the rendering issue - I set the widget visibility to False before deleting it. If this fixes the problem, I think it could be merged, we've been looking at it for two weeks almost. Your call, I guess.

**Comment by m-kuhn** [comment]
Thank you.

I'm quite confident that it works stable, but given that there has no broad testing been done, there are only 3 days left and there have been no complaints about the current state I think we should just postpone it to next week when feature freeze is lifted.

**Comment by m-kuhn** [comment]
Wooh, it feels great to use it!
Do you think it would be hard to make the dropdown stay if the filter column / filter style is not changed?

**Comment by carolinux** [comment]
You mean to not be replaced with the plain line  edit that exposes the expression once it has been applied?

**Comment by m-kuhn** [comment]
Exactly

**Comment by carolinux** [comment]
Heh, this is *even* nicer to use

**Comment by m-kuhn** [comment]
Merged in eff1357e2b6

**Pull request closed without merging**